File: server/service/follow.go
package service

import (
	"kuukaa.fun/leaf/domain/model"
)

// 关注
func InsertFollow(follow model.Follow) error {
	return mysqlClient.Create(&follow).Error
}

// 取消关注
func DeleteFollow(uid uint, fid uint) error {
	return mysqlClient.Where("uid = ? and fid = ?", uid, fid).Delete(&model.Follow{}).Error
}

// 获取关注状态
func IsFollow(uid uint, fid uint) bool {
	var follow model.Follow
	mysqlClient.Where("uid = ? and fid = ?", uid, fid).First(&follow)

	return follow.ID != 0
}

// 通过UID获取关注列表
func SelectFollowingUser(userId uint, page, pageSize int) (users []model.User) {
	offset := (page - 1) * pageSize
	userIds := mysqlClient.Model(&model.Follow{}).Select("fid").Where("uid = ?", userId)
	mysqlClient.Where("id in (?)", userIds).Limit(pageSize).Offset(offset).Find(&users)
	return
}

// 通过UID获取粉丝列表
func SelectFollowerUser(userId uint, page int, pageSize int) (users []model.User) {
	offset := (page - 1) * pageSize
	userIds := mysqlClient.Model(&model.Follow{}).Select("uid").Where("fid = ?", userId)
	mysqlClient.Where("id in (?)", userIds).Limit(pageSize).Offset(offset).Find(&users)
	return
}

// 通过UID获取粉丝数
func GetFollowCount(uid uint) (following, follower int64) {
	mysqlClient.Model(&model.Follow{}).Where("uid = ?", uid).Count(&following)
	mysqlClient.Model(&model.Follow{}).Where("fid = ?", uid).Count(&follower)
	return
}

File: web/apis/src/api/follow.ts
import request from '../request'

//关注
export const followAPI = (id: number) => {
    return request.post('v1/follow/add', { id });
}

//取消关注
export const unfollowAPI = (id: number) => {
    return request.post('v1/follow/cancel', { id })
}

//获取关注状态
export const getFollowStatusAPI = (fid: number) => {
    return request.get(`v1/follow/status?fid=${fid}`)
}

//获取关注数据
export const getFollowDataAPI = (uid: number) => {
    return request.get(`v1/follow/count?uid=${uid}`)
}

//获取关注列表
export const getFollowingAPI = (uid: number, page: number, page_size: number) => {
    return request.get(`v1/follow/following?uid=${uid}&page=${page}&page_size=${page_size}`)
}

//获取粉丝列表
export const getFollowersAPI = (uid: number, page: number, page_size: number) => {
    return request.get(`v1/follow/follower?uid=${uid}&page=${page}&page_size=${page_size}`)
}